### ISSUE TITLE
Issue with toggling between galleries on same page

### ISSUE THREAD

**Issue by g1st** [xiaolin/react-image-gallery#422]
If a page has more than one gallery when toggling between them `currentIndex` still points to the previous one. In example below this causes second gallery to appear without any image. Because it has only one item and index `1` does not exist.

![Issue example](https://media.giphy.com/media/TjvjnVmkN4sbIjzyRB/giphy.gif)

[Example repo](https://github.com/g1st/react-image-gallery-flaw).

I have fixed it in `componentDidUpdate` adding this:
```
   if (itemsChanged) {
      this._handleResize();
      this.setState({ currentIndex: this.props.startIndex });
    }
```

But its not ideal as newly mounted gallery will still inherit `currentIndex` from previous one when `items.length` is same. 

I assume it's a flaw or is it expected behaviour and I'm doing something fundamentally wrong here?

**Comment by xiaolin** [comment]
For `this.props.startIndex` how are you handling that? For each gallery you toggle through, you should be keeping track of the start index.

e.g. a rough example might look like
```jsx
const galleryProps = {
  'gallery1': {
     startIndex: 1,
  },
  'gallery2': {
    startIndex: 2,
  }
}

render() {
  return <ImageGallery ...galleryProps['gallery1'] />
}
```

**Comment by g1st** [comment]
Sorry but it is still the same, `<ImageGallery/>`'s `currentIndex` does not update when toggling them. It gets the right `startIndex` prop value but state does not update.

I updated my example as your suggestion [here](https://github.com/g1st/react-image-gallery-flaw/blob/master/src/App.js).

I'm not setting `startIndex` in my other repo as I want it to be default value of `0`. But when switching galleries on same page they don't remount and inherit `currentIndex` from previously loaded gallery. So only forcing `currentIndex` to change when items.length are not the same (code sample in my original post) forces state to update. This is directly in react-image-gallery package in `node_modules` just a work around for now.

**Comment by xiaolin** [comment]
Can you show me an example using codesandbox.io so I can verify the issue?

**Comment by g1st** [comment]
Here is [code sanbox example](https://codesandbox.io/embed/stoic-edison-8sbnz).

**Comment by xiaolin** [comment]
Thank you, looks to be a bug as new currentIndex prop is not respected.

**Comment by xiaolin** [comment]
fixed in commit `52c0cc7`
